src/aura/CRLP_Rollup/CRLP_RollupHelper.js

```diff
@@ -63,278 +82,643 @@
+            rollupType.name = 'Allocation__c';
+            rollupType.summaryObject = 'General_Accounting_Unit__c';
+            rollupType.label = labels.allocationLabel + ' -> ' + labels.gauLabel;
         }
-        //reset field to null regardless of summary object value
-        cmp.set("v.activeRollup.Summary_Field__r.QualifiedApiName",null);
+
+        cmp.set("v.selectedRollupType", rollupType);
+        console.log(cmp.get("v.selectedRollupType"));
     },
 
-    resetRollup: function(cmp) {
+    setIntegerYearList: function(cmp){
+        var integerList = [];
+        var timeLabel = cmp.get("v.selectedYearlyOperationLabel");
 
-        //TODO: refactor this: consider caching rolllup, and review what is necessary
-        var activeRollupId = cmp.get("v.activeRollup.Id");
-        var action = cmp.get("c.getRollupById");
-        action.setParams({id: activeRollupId});
+        integerList.push({name: 0, label: 'This Year'});
+        integerList.push({name: 1, label: 'Last Year'});
+        for(var i=2; i<21; i++) {
+            integerList.push({name: i, label: i + ' Years Ago'});
+        }
 
-        action.setCallback(this, function (response) {
-            var state = response.getState();
-            if (state === "SUCCESS") {
-                var response = response.getReturnValue();
-                cmp.set("v.activeRollup", response);
-                var activeRollupMap = cmp.get("v.activeRollup");
+        cmp.set("v.integerList", integerList);
+    },
 
-                //need to reset all fields to match the selected objects
-                this.resetAllFields(cmp);
+    updateAllowedOperations: function(cmp){
 
-            }
-            else if (state === "ERROR") {
-                var errors = response.getError();
-                if (errors) {
-                    if (errors[0] && errors[0].message) {
-                        console.log("Error message: " +
-                            errors[0].message);
-                    }
-                } else {
-                    console.log("Unknown error");
-                }
-            }
-        });
-        $A.enqueueAction(action);
-    },
+        //start with all operations
+        var ops = cmp.get("v.operations");
+        console.log(JSON.stringify(ops));
 
-    resetOperationFieldLabel: function(cmp, operation, operationList){
-        var label;
-        for(var i=0; i<operationList.length; i++){
-            if(operationList[i].name == operation){
-                label = operationList[i].label;
+        var field = cmp.get("v.activeRollup.Summary_Field__r.QualifiedApiName");
+        var summaryFields = cmp.get("v.summaryFields");
+
+        var type;
+        for (var i = 0; i < summaryFields.length; i++) {
+            if (summaryFields[i].name === field) {
+                type = summaryFields[i].type;
                 break;
             }
         }
-        return label;
+        console.log(type);
```

Comment:
Testing this in a scratch org and found that for the "GAU: Count of Allocation Last Year" rollup, the Operation field is blank on the View and Edit pages. This appears to be because the `type` var is undefined, so the code below isn't able to include the various operations for the numeric field. I've noticed that Operation is blank on other 'View' pages as well, but shows up when edit is clicked.

Reply:
Thanks for the reminder on this! It did get fixed on the next branch, but it required a few changes that I'd like to keep separate since they were part of a larger change.